### ISSUE TITLE
About Page - CSS to Hide/Show Section 

### ISSUE THREAD

**Issue by suziebytes** [suziebytes/Resswell#9]
Write code that will: 
- start with 'about' section set to visible 
- hide any sections that are visible, if it's link is not clicked
- will need to keep track of link that is currently clicked 
- will need to toggle sections to show/hide

#struggles 
this is what I get for going on hiatus 👎

**Comment by suziebytes** [comment]
Ended up creating a three column grid system called, `.tabs` with buttons in each column.

```css
.tabs {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-auto-rows: 40px;
    font-family: arial;
    color: white;
    font-size: 11pt;
    font-weight: 50;
    text-align: center;
    opacity: .6;
    margin-left: 40%;
    margin-right: 20px;
    margin-top: 4%;
}
```
```html 
 <main class="about">
        <div class="tabs">
            <button id="about">ABOUT
                <!-- <div id="aboutext">
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Autem laborum animi deserunt, quae
                    placeat
                    eligendi sint illum mollitia consequuntur debitis nihil ipsam reprehenderit voluptates porro
                    nisi
                    nesciunt quia. Repudiandae, neque?
                </div> -->
            </button>

            <button id="services">SERVICES
                <div id="aboutservices">
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Autem laborum animi deserunt, quae
                    placeat
                    eligendi sint illum mollitia consequuntur debitis nihil ipsam reprehenderit voluptates porro
                    nisi
                    nesciunt quia. Repudiandae, neque?
                </div>
            </button>
            <button id="why">WHY RESSWELL</button>
        </div>


    </main>
```

This created: 
![image](https://user-images.githubusercontent.com/32041993/52838666-529a6580-30a8-11e9-9e3e-ddd9d8941390.png)

**Comment by suziebytes** [comment]
Instead of clicking on each `tab` I created a `hover` function that would make the text visible or not visible. 

The text for each tab section is initially set to be hidden using css `display: none` 
```css

#aboutext,
#aboutservices {
    display: none;
}

```
but upon `hover`, the text section in css will `display: block`

```css

#services:hover #aboutservices {
    display: block;
}

#aboutservices {
    height: 200px;
    width: 400px;
    color: grey;
    background-color: white;
    opacity: .5;
}
```

This ended up resulting in: 
![image](https://user-images.githubusercontent.com/32041993/52838822-c89ecc80-30a8-11e9-8bd4-e1923b647215.png)



Current Issues: 

- Section text seems to be dependent on the button and the size of the text width/height also enlarges the button size 
- Text is currently sitting on top of the button 
- Text needs to be formatted properly 
 
Possible resolution: 
Put the `tab` in it's own three column grid 
Create a new <div> container to contain all 3 section text 
Format <div> container to the w/h I want it to appear on the page 
Edit the `hover` CSS so that it shows the new div with it's specific section text

**Comment by suziebytes** [comment]
Was able to successfully make it better, but the work is still not done. 
 
Created generic class/id names for the tab and the content and nested them within a container.  
`display: none` the content until the tab was `hover`-ed over and then we changed it to `display: block` 
This is what the structure looked like: 
![image](https://user-images.githubusercontent.com/32041993/52890828-bde14780-313b-11e9-824b-bcffd7b301c2.png)

This is what the site looked like: 
<img width="715" alt="screen shot 2019-02-15 at 12 22 26 pm" src="https://user-images.githubusercontent.com/32041993/52890764-6a6ef980-313b-11e9-974b-9462a725db05.png">

Issue that I am currently having now is that the content is only visible to the width of the column. 
When adjusting the content width, the tab will also adjust as well since they are nested together. 

![unknown](https://user-images.githubusercontent.com/32041993/52890807-95594d80-313b-11e9-97da-11c6526b1f7b.png)


**Possible Resolution** 
Creating 3 columns and 2 rows 
row 1 will be the tabs 
row 2 will be the content 

The issue with this aside from starting over is that I might have the select each content specifically using classes, where as the previous code allowed me to use generic classes to select everything. 

For example, 
`tab` vs `abouttab` 
`content` vs `aboutcontent` 

boo.... 🐱

**Comment by suziebytes** [comment]
Ending up talking to some people and they said I was overcomplicating this and *forcing* the use of flexbox and grid. 
 
I had re-written this multiple times and got certain things to work, but in reality using flex/grid wasn't necessary. I ended up scrapping it and redoing this page. It really only needed an about section, which I've included. The others were just filler tabs with the sole intent of using grid/flexbox.
